packages/react-storage/src/components/StorageBrowser/views/LocationActionView/CreateFolderControls.tsx

```diff
@@ -77,31 +82,41 @@ export const CreateFolderControls = (): React.JSX.Element => {
     handleCreateAction({ prefix: '', options: { reset: true } });
   };
 
-  const primaryProps =
+  const primaryProps: ActionStartProps =
     result?.status === 'COMPLETE'
       ? {
           onClick: () => {
             handleClose();
           },
-          children: 'Folder created',
+          label: 'Folder created',
         }
       : {
           onClick: () => {
             handleCreateFolder();
           },
-          children: 'Create Folder',
+          label: 'Create Folder',
           disabled: !folderName || !!fieldValidationError,
         };
```

Comment:
Let's just remove the "primary" naming altogether and move this into the contextValue below directly.

Also think we can save ourselves a bit of overhead and have just
```ts
onClick: handleClose
...
onClick: handleCreateFolder
```